### ISSUE TITLE
Python ImportError ModuleNotFoundError in Ubuntu

### ISSUE THREAD

**Issue by anacsousa1** [opensim-org/opensim-core#2827]
Hi, I am trying to install OpenSim in Ubuntu 18.04, but I am getting `ModuleNotFoundError: No module named 'opensim'` when trying to import opensim through python.

Versions installed:

- Ubuntu 18.04 64bit
- cmake 3.10.2
- g++7.5.0
- python 3.6.9
- swig 4.0.2

Everything compiles without errors. However, when running the tests, it fails for python_examples and python_tests.
I attached the error output when running `ctest --output-on-failure -R python_tests`: [error_python_ctest.txt](https://github.com/opensim-org/opensim-core/files/4881409/error_python_ctest.txt)

I've seen solutions that mentioned some problems with the paths for python (cmake). However, I am not sure where would be wrong in mine:
```
//Path to a program.
PYTHON_EXECUTABLE:FILEPATH=/usr/bin/python3

//Path to a file.
PYTHON_INCLUDE_DIR:PATH=/usr/include/python3.6m

//Path to a library.
PYTHON_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libpython3.6m.so

//Path to a library.
PYTHON_LIBRARY_DEBUG:FILEPATH=PYTHON_LIBRARY_DEBUG-NOTFOUND
```

I have also tried to set the LD_LIBRARY_PATH as:
`export LD_LIBRARY_PATH=:~/opensim-core/lib`
But it was not successful.

What I am missing here?

**Comment by chrisdembia** [comment]
Hey @anacsousa1, thanks for attaching the log. The key message is `ImportError: libSimTKmath.so.3.7: cannot open shared object file: No such file or directory`.

I suspect the issue is how you set `LD_LIBRARY_PATH`. Are the OpenSim libraries indeed located at `~/opensim-core/lib`, or are they in a `x86_64-linux-gnu` subdirectory?

**Comment by anacsousa1** [comment]
I am not entirely sure, but I believe they are in opensim-core/lib:
```

ana@ana-lenovo-ubuntu:~/opensim-core/lib$ ls
cmake                        libosimLepton.so       libSimTKmath.so
libosimActuators.so          libosimSimulation.so   libSimTKmath.so.3.7
libosimAnalyses.so           libosimTools.so        libSimTKsimbody.so
libosimCommon.so             libSimTKcommon.so      libSimTKsimbody.so.3.7
libosimExampleComponents.so  libSimTKcommon.so.3.7  python3.6
```

Is that it?

**Comment by aymanhab** [comment]
SWIG 4.0 is not supported.

**Comment by chrisdembia** [comment]
Ah yes, install `swig3.0` instead. I do not know how you successfully built the bindings if you used SWIG 4.0.

**Comment by anacsousa1** [comment]
Hi @aymanhab and @chrisdembia, thanks for the answer.
I installed both swig3 and 4, I just checked the CMakeCache again that I am building for swig3.0:

```
//Path to a file.
SWIG_DIR:PATH=/usr/share/swig3.0

//Path to a program.
SWIG_EXECUTABLE:FILEPATH=/usr/bin/swig3.0

//Swig version
SWIG_VERSION:STRING=3.0.12
```

**Comment by aymanhab** [comment]
The error log refers to /home/ana/opensim-core-build/Bindings/Python/RelWithDebInfo/opensim/ which is a build folder rather than an install folder, maybe that's the issue with LD_LIBRARY_PATH. You should install first and then import the opensim package from the install location. Not sure how the test case is setup to handle this, but the github action builds and tests with python 3.7 so you may take a look there to compare notes/logs: 
https://github.com/opensim-org/opensim-core/runs/843439791?check_suite_focus=true

**Comment by Justinyu1618** [comment]
I'm getting this exact same issue. @anacsousa1 were you able to find a solution?

**Comment by anacsousa1** [comment]
I got some help from a colleague, and we were able to fix it by:
1. Running the setup.py for opensim from the build directory
```
cd /opensim-core/lib/python3.6/site-packages
sudo python3 setup.py install
```
2. Adding the library directory to the path
`export LD_LIBRARY_PATH=~/opensim-core/lib`
3. Making python3 the default
`sudo update-alternatives --install /usr/bin/python python /usr/bin/python3`

As it is fixed, I am closing the issue. Thanks for the comments and suggestions.
